core/src/main/scala/kafka/raft/RaftManager.scala

```diff
@@ -128,9 +128,15 @@ class KafkaRaftManager(
   def metaLogManager: MetaLogManager = metaLogShim
 
   def startup(): Unit = {
+    // Wait for the controller quorum voters string to be set
+    raftConfig.updateQuorumVoters(controllerQuorumVotersFuture.get())
```

Comment:
Related to one of my other comments, but do we really need to do this? I think we should treat `RaftConfig` as immutable and only specifying the initial configuration. We can say that `NetworkChannel` is the source of truth for the current endpoint information of the voters.

Reply:
Yea, fair enough. I was going off of the expectation that the RaftConfig would stay consistent with the current cluster state. But this would mean that the RaftConfig is more of a source of truth what the cluster was configured with at startup, if that makes sense. Fixed.